teamsupport/models.py

```diff
@@ -23,6 +27,27 @@ def __init__(self, client, ticket_id=None, data=None):
                 '(neither given)')
         self.id = self.TicketID
 
+    @classmethod
+    def create(cls, client, first_name, email, category, name, description):
+        # We need to associate ticket with Contact, otherwise ticket doesn't
+        # make sense. First, we try to find an existing contact.
+        contact = Contact.get(client, first_name, email)
+        if contact is None:
+            # Otherwise - create new one.
+            contact = Contact.create(client, first_name, email)
```

Comment:
it exists because the pattern of get, check null, if so then create is quite common :)